test/test_grid.py

```diff
@@ -196,8 +197,63 @@ def test_integrate(self):
 
 class TestFaceAreas(TestCase):
 
+    def test_calculate_total_face_area_triangle(self):
+        """Create a uxarray grid from vertices and calculate the area using
+        gaussian and triangular quadrature rules."""
+
+        verts = np.array([[0.57735027, -5.77350269e-01, -0.57735027],
+                          [0.57735027, 5.77350269e-01, -0.57735027],
+                          [-0.57735027, 5.77350269e-01, -0.57735027]])
+        vgrid = ux.Grid(verts)
+
+        # get node names for each grid object
+        x_var = vgrid.ds_var_names["Mesh2_node_x"]
+        y_var = vgrid.ds_var_names["Mesh2_node_y"]
+        z_var = vgrid.ds_var_names["Mesh2_node_z"]
+
+        vgrid.ds[x_var].attrs["units"] = "m"
+        vgrid.ds[y_var].attrs["units"] = "m"
+        vgrid.ds[z_var].attrs["units"] = "m"
+
+        area_gaussian = vgrid.calculate_total_face_area(
+            quadrature_rule="gaussian", order=5)
+        nt.assert_almost_equal(area_gaussian, constants.TRI_AREA, decimal=3)
+
+        area_triangular = vgrid.calculate_total_face_area(
+            quadrature_rule="triangular", order=4)
+        nt.assert_almost_equal(area_triangular, constants.TRI_AREA, decimal=1)
+
+    def test_calculate_total_face_area_file(self):
+        """Load a grid from file and calculate the total face area of the
+        mesh."""
+
+        xr_grid = xr.open_dataset(str(mesh_file30))
+        grid = ux.Grid(xr_grid)
+        area = grid.calculate_total_face_area()
+        nt.assert_almost_equal(area, constants.MESH30_AREA, decimal=3)
+
+    def test_compute_face_areas(self):
+        """Load a grid from file and calculate the area of each face of the
+        mesh."""
+
+        xr_grid = xr.open_dataset(str(mesh_file30))
+        grid = ux.Grid(xr_grid)
+        # by default triangular quadrature is used and order is set to 4th.
+        area = grid.compute_face_areas()
+
+        assert (area.size == grid.ds.nMesh2_face.size)
```

Comment:
```Python
assert (area.size == grid.nMesh2_face)

Reply:
Right, waiting for the new design for updating this old PR.

Reply:
test_grid has gotten really big, we should think about making in more modular in another PR.